Tests/NIOTests/HappyEyeballsTest.swift

```diff
@@ -32,8 +32,9 @@ private extension Array where Element == Channel {
         self.forEach {
             do {
                 _ = try($0 as! EmbeddedChannel).finish()
+                // We're happy with no error
             } catch ChannelError.alreadyClosed {
-                return
+                return // as well as already closed.
```

Comment:
Is there any value in something like `assertMaybeThrows` which validates the error if one is thrown but doesn't fail if no error is thrown? I think I've only seen one or two occurrences of this elsewhere here so probably not...

Reply:
There are a few but unfortunately our `TestUtils.swift` doesn't really scale. It's always just available in one test module but we have lots of modules and like 5 different repos :|. So I now prefer to stick with plain XCTest whenever possible...